router/http/src/main/scala/io/buoyant/router/http/TracingFilter.scala

```diff
@@ -1,43 +1,86 @@
 package io.buoyant.router.http
 
-import com.twitter.finagle.{Service, SimpleFilter}
-import com.twitter.finagle.http.{Request, Response}
+import com.twitter.conversions.storage._
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack, Stackable, param}
+import com.twitter.finagle.client.StackClient
+import com.twitter.finagle.http.{Request, Response, Status}
 import com.twitter.finagle.tracing.Trace
 
+object TracingFilter {
+  val role = StackClient.Role.protoTracing
+  val module: Stackable[ServiceFactory[Request, Response]] =
+    new Stack.Module1[param.Tracer, ServiceFactory[Request, Response]] {
+      val role = TracingFilter.role
+      val description = "Traces HTTP-specific request metadata"
+      def make(_tracer: param.Tracer, next: ServiceFactory[Request, Response]) = {
+        val param.Tracer(tracer) = _tracer
+        if (tracer.isNull) next
+        else (new TracingFilter).andThen(next)
+      }
+    }
+
+  val TransferEncoding = "transfer-encoding"
+
+  val MaxBodySize = 64 * 1000 // 64KB-ish
+}
+
 /**
  * Annotates HTTP method, uri, and status code, content-type, and content-length.
  */
-object TracingFilter extends SimpleFilter[Request, Response] {
+class TracingFilter extends SimpleFilter[Request, Response] {
+  import TracingFilter.{TransferEncoding, MaxBodySize}
 
   def apply(req: Request, service: Service[Request, Response]) = {
     recordRequest(req)
     service(req).onSuccess(recordResponse)
   }
 
-  private[this] def recordRequest(req: Request): Unit = {
-    Trace.recordRpc(req.method.toString)
-    // http.uri is used here for consistency with finagle-http's tracing filter
-    Trace.recordBinary("http.uri", req.uri)
-    Trace.recordBinary("http.req.method", req.method.toString)
-    req.host.foreach(Trace.recordBinary("http.req.host", _))
-    Trace.recordBinary("http.req.version", req.version.toString)
-    req.contentLength.foreach(Trace.recordBinary("http.req.content-length", _))
-    req.contentType.foreach(Trace.recordBinary("http.req.content-type", _))
-    req.headerMap.get("transfer-encoding").foreach { te =>
-      Trace.recordBinary("http.req.transfer-encoding", te)
+  private[this] def recordRequest: Request => Unit =
```

Comment:
you're right good catch. it should be a val. and it can be a method but kl asked me to change it.  I think it makes sense as a method, so I'll change it.